Looking at the [failing windows tests](https://github.com/biojppm/rapidyaml/actions/runs/26492676288/job/78078892474?pr=610#step:5:4933) : the newly-added  `C4_NORETURN` to the tree handler events now causes a `warning C4702: unreachable code` on MSVC.

This is triggered from an engine test that checks for an error (ie requires an error) in the specified sequence of event handler calls when using the tree handler. That test must remain.

IMO the way to tackle this is to work around it by suppressing that warning it in the offending test source code: adding `C4_SUPPRESS_WARNING_MSVC_WITH_PUSH(4702)` and corresponding `_POP` at the top/bottom of the files. 

AFAICT these are the files where that would be needed:

  - test/test_engine_3_map_block_2.cpp
  - test/test_engine_3_map_flow_1.cpp
  - test/test_engine_4_anchor.cpp
  - test/test_engine_5_tag_1.cpp
  - test/test_engine_6_qmrk_1.cpp
  - test/test_engine_6_qmrk_2.cpp
  - test/test_engine_6_qmrk_3.cpp
  - test/test_engine_6_qmrk_4.cpp
  - test/test_engine_7_seqimap.cpp

HTH.

--- a/src/c4/yml/common.hpp
+++ b/src/c4/yml/common.hpp
@@ -289,7 +289,7 @@ struct RYML_EXPORT Location
 
     operator bool () const noexcept { return !name.empty() || line != npos || offset != npos || col != npos; }
 
-    C4_NO_INLINE Location() noexcept : offset(npos), line(npos), col(npos), name() {};
+    C4_NO_INLINE Location() noexcept : offset(npos), line(npos), col(npos), name() {}
     C4_NO_INLINE Location(                         size_t l          ) noexcept : offset(npos), line(l), col(npos), name() {}
     C4_NO_INLINE Location(                         size_t l, size_t c) noexcept : offset(npos), line(l), col(c   ), name() {}
     C4_NO_INLINE Location(               size_t b, size_t l, size_t c) noexcept : offset(b   ), line(l), col(c   ), name() {}

--- a/src/c4/yml/node.hpp
+++ b/src/c4/yml/node.hpp
@@ -725,7 +725,7 @@ public:
     {
         _C4RR();
         return from_chars(val(), &v);
-    };
+    }
 
     /** @} */
 

--- a/src/c4/yml/parse_engine.def.hpp
+++ b/src/c4/yml/parse_engine.def.hpp
@@ -1134,6 +1134,7 @@ bool ParseEngine<EventHandler>::_scan_scalar_plain_seq_flow(ScannedScalar *C4_RE
                     _c4err("parse error"); // no return
                 }
             }
+            break;
         default:
             ;
         }

--- a/src/c4/yml/parse_engine.def.hpp
+++ b/src/c4/yml/parse_engine.def.hpp
@@ -463,7 +463,8 @@ void ParseEngine<EventHandler>::_relocate_arena(csubstr prev_arena, substr next_
     if((s).str >= pb && (s).str <= pe)              \
     {                                               \
         (s).str = next_arena.str + ((s).str - pb);  \
-    }
+    }                                               \
+    ((void)0)
     for(ParserState &st : m_evt_handler->m_stack)
     {
         _ryml_relocate(st.line_contents.rem);
@@ -3584,11 +3585,7 @@ void ParseEngine<EventHandler>::_filter_block_folded_indented_block(FilterProces
                 {
                     const char c = rem[first];
                     _c4dbgfbf("firstns={}='{}'", first, _c4prc(c));
-                    if(c == '\n' || c == '\r')
-                    {
-                        ;
-                    }
-                    else
+                    if(c != '\n' && c != '\r')
                     {
                         _c4dbgfbf("done with indented block",  first);
                         goto endloop;

--- a/src/c4/yml/parse_engine.def.hpp
+++ b/src/c4/yml/parse_engine.def.hpp
@@ -1616,7 +1616,7 @@ substr ParseEngine<EventHandler>::_peek_next_line(size_t pos) const
 
 next_is_empty:
     _c4dbgpf("peek next line @ {}: (len=0)''", pos);
-    return {};
+    return rem;
 }
 
 //-----------------------------------------------------------------------------

--- a/src/c4/yml/tag.hpp
+++ b/src/c4/yml/tag.hpp
@@ -130,10 +130,10 @@ struct RYML_EXPORT TagDirectives
     void clear() noexcept;
     id_type size() const noexcept;
     TagDirective const* lookup(csubstr tag, id_type id) const noexcept;
-    TagDirective * begin() noexcept { return m_directives; };
-    TagDirective * end() noexcept { return m_directives + size(); };
-    TagDirective const* begin() const noexcept { return m_directives; };
-    TagDirective const* end() const noexcept { return m_directives + size(); };
+    TagDirective * begin() noexcept { return m_directives; }
+    TagDirective * end() noexcept { return m_directives + size(); }
+    TagDirective const* begin() const noexcept { return m_directives; }
+    TagDirective const* end() const noexcept { return m_directives + size(); }
     TagDirectiveRange directives() const noexcept { return TagDirectiveRange{m_directives, m_directives + size()}; }
     TagDirectiveRange lookup_range(id_type doc_id) const noexcept;
     /** @note the str member of the return value may be null, meaning

--- a/src/c4/yml/tree.hpp
+++ b/src/c4/yml/tree.hpp
@@ -1442,7 +1442,7 @@ template<class T>
 size_t to_chars_float(substr buf, T val)
 {
     static_assert(std::is_floating_point<T>::value, "must be floating point");
-    C4_SUPPRESS_WARNING_GCC_CLANG_WITH_PUSH("-Wfloat-equal");
+    C4_SUPPRESS_WARNING_GCC_CLANG_WITH_PUSH("-Wfloat-equal")
     if(C4_UNLIKELY(std::isnan(val)))
         return to_chars(buf, csubstr(".nan"));
     else if(C4_UNLIKELY(val == std::numeric_limits<T>::infinity()))

